dataprocspawner/spawner.py

```diff
@@ -801,12 +938,23 @@ def change_zone_in_cluster_cfg(self, cluster_data):
       else:
         new_zone = current_zone_tmp
 
-    cluster_data['config']['gce_cluster_config']['zone_uri'] = f'https://www.googleapis.com/' \
-                                                              f'compute/v1/projects/' \
-                                                              f'{self.project}/zones/' \
-                                                              f'{new_zone}'
+    cluster_data['config']['gce_cluster_config']['zone_uri'] = (
+        'https://www.googleapis.com/compute/v1/projects/'
+        f'{self.project}/zones/{new_zone}')
     return cluster_data
 
+  async def get_cluster_notebook_endpoint(self, clustername):
+    """ Extracts Component Gateway endpoint for Jupyter on the cluster.
+
+    Returns:
+      (str) Component Gateway URL without the trailing `/`
+    """
+    cluster = await self.get_cluster(clustername)
+    if cluster is not None:
+      endpoint_name = 'JupyterLab' if 'lab' in self.default_url else 'Jupyter'
```

Comment:
Let's make a note to deprecate the default_url at some point: I think we're pretty much standardizing on JupyterLab, and it doesn't really make sense to have this now that we're using Component Gateway anyways.

Reply:
Component Gateway can use to /jupyter or jupyter/lab similar to Jupyter.
Agreed that we can always use JupyterLab but because there is the option in JupyterHub and CG, I thought of keeping it.